var/spack/repos/builtin/packages/dia/package.py

```diff
@@ -34,15 +34,18 @@ class Dia(Package):
 
     depends_on('intltool', type='build')
     depends_on('gtkplus@2.6.0:')
-    depends_on('cairo')
-    depends_on('libpng')
```

Comment:
See below. If `dia` depends on `cairo` and `libpng`, it should declare it as a direct dependency. We shouldn't rely on transitive dependencies to always be in RPATH.